runtime/lua/vim/text.lua

```diff
@@ -2,14 +2,27 @@
 
 local M = {}
 
+local alphabet = '0123456789ABCDEF'
+local lookup = {} ---@type table<integer|string, integer|string>
+do
+  for i = 1, #alphabet do
+    local char = alphabet:sub(i, i)
+    lookup[i - 1] = char
+    lookup[char] = i - 1
+    lookup[char:lower()] = i - 1
+  end
+end
+
 --- Hex encode a string.
 ---
 --- @param str string String to encode
 --- @return string : Hex encoded string
 function M.hexencode(str)
   local enc = {} ---@type string[]
   for i = 1, #str do
-    enc[i] = string.format('%02X', str:byte(i, i + 1))
+    local byte = str:byte(i)
+    enc[#enc + 1] = lookup[math.floor(byte / 16)] --[[@as string]]
+    enc[#enc + 1] = lookup[byte % 16] --[[@as string]]
```

Comment:
Would be curious to know if calculating the correct value of `i` (and avoiding `#enc`) (like in `hexdecode()`) makes a difference here?